src/main/java/net/conjur/api/clients/AuthnClient.java

```diff
@@ -16,40 +22,67 @@
  * which can then be used to make authenticated calls to other conjur services.
  *
  */
-public class AuthnClient extends ConjurClient implements AuthnProvider {
+public class AuthnClient implements AuthnProvider {
 
     private WebTarget login;
     private WebTarget authenticate;
 
+    private final Endpoints endpoints;
+
+    private String apiKey;
+
 	public AuthnClient(final String username, final String password, final Endpoints endpoints) {
-	    super(username, password, endpoints);
+        this.endpoints = endpoints;
 
-        init(username);
-	}
+        init(username, password);
+
+        // replacing the password with an API key
+        this.apiKey = login();
+    }
 
-    public Token authenticate(String apiKey) {
+    public Token authenticate() {
 	    Response res = authenticate.request("application/json").post(Entity.text(apiKey), Response.class);
 	    validateResponse(res);
 
         return Token.fromJson(res.readEntity(String.class));
      }
 
     // implementation of AuthnProvider method
-    public Token authenticate(String apiKey, boolean useCachedToken) {
-        return authenticate(apiKey);
+    public Token authenticate(boolean useCachedToken) {
+        return authenticate();
     }
 
+    /**
+     * Login to a Conjur account with the credentials specified in the configuration
+     * @return The API key of the user
+     */
     public String login(){
 	    Response res = login.request("text/plain").get(Response.class);
 	    validateResponse(res);
 
         return res.readEntity(String.class);
      }
 
-    private void init(final String username){
-        WebTarget root = getClient().target(getEndpoints().getAuthnUri());
+    private void init(final String username, final String password){
+        final ClientBuilder builder = ClientBuilder.newBuilder()
+                .register(new HttpBasicAuthFilter(username, password));
+
+        HostNameVerification.getInstance().updateClientBuilder(builder);
+
+        Client client = builder.build();
+        WebTarget root = client.target(endpoints.getAuthnUri());
+
         login = root.path("login");
         authenticate = root.path(encodeUriComponent(username)).path("authenticate");
     }
 
+    // TODO orenbm: Remove when we have a response filter to handle this
+    private void validateResponse(Response response) {
+        int status = response.getStatus();
+        if (status < 200 || status >= 400) {
```

Comment:
I think this is fine for now, but in the future we should probably explicitly define expected responses.  In general, they should only include `200` or `201`.